spec/datadog/appsec/rate_limiter_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/rate_limiter'
+
+RSpec.describe Datadog::AppSec::RateLimiter do
+  before { described_class.reset!(:traces) }
+
+  describe '#limit' do
+    context 'in different threads' do
+      before { stub_const("#{described_class}::TRACES_KEY", :__spec_traces) }
+
+      it 'creates separate rate limiter per thread' do
+        thread_1 = Thread.new do
+          described_class.limit(:traces) { 'workload' }
+        end
+
+        thread_2 = Thread.new do
+          described_class.limit(:traces) { 'workload' }
+        end
+
+        thread_1.join
+        thread_2.join
+
+        rate_limiter_1 = thread_1.thread_variable_get(:__spec_traces)
+        rate_limiter_2 = thread_2.thread_variable_get(:__spec_traces)
+
+        expect(rate_limiter_1.object_id).not_to eq(rate_limiter_2.object_id)
```

Comment:
Minor: I think this is equivalent?

```suggestion
        expect(rate_limiter_1).not_to be(rate_limiter_2)
```